crates/extensions/c8y_mapper_ext/src/operations/handlers/custom_operation.rs

```diff
@@ -31,10 +34,36 @@ impl OperationContext {
                 extra_messages: vec![],
             }),
             CommandStatus::Successful => {
-                let smartrest_set_operation = self.get_smartrest_successful_status_payload(
-                    CumulocitySupportedOperations::C8yCustom(operation_name.to_string()),
-                    cmd_id,
-                );
+                let operation =
+                    CumulocitySupportedOperations::C8yCustom(operation_name.to_string());
```

Comment:
I understand now the issue highlighted by this comment: https://github.com/thin-edge/thin-edge.io/pull/3283/files#r1876437428

The `OperationContext` gives no context about the c8y operation but only about the tedge operation.

As suggested offline by @rina23q:
- the C8y name of the operation can be injected into the tedge operation on init.
- and extracted back here when the tedge operation is finalized.

However, to implement what is suggested here https://github.com/thin-edge/thin-edge.io/pull/3283#issuecomment-2535137925, more context is required. One would need the whole c8y custom operation definition or at least the except to be used for the response.

-

Reply:
To proceed Albin's proposal, `OperationHandler` (aka `OperationContext`) must hold custom handlers. Once it has done, indeed the issue of c8y's operation name can be solved.

Reply:
Probably the quick fix is, to move `supported_operation` from `CumulocityConverter` to `OperationHandler`.
Then `supported_operation` can be updated by `CumulocityConverter` in runtime, and also `OperationHandler` can access it.

```rust
pub struct CumulocityConverter {
   ...
    // supported_operations: SupportedOperations, // remove from here
    pub operation_handler: OperationHandler,
}

```

```rust
pub struct OperationHandler {
    context: Arc<OperationContext>,
    pub supported_operations: SupportedOperations,  // added
    running_operations: HashMap<Arc<str>, RunningOperation>,
}
```